packages/server/src/fhir/repo.ts

```diff
@@ -2327,6 +2327,11 @@ async function getCacheEntry<T extends Resource>(resourceType: string, id: strin
  * @returns Array of cache entries or undefined.
  */
 async function getCacheEntries(references: Reference[]): Promise<(CacheEntry<Resource> | undefined)[]> {
+  const referenceKeys = references.map((r) => r.reference as string);
+  if (referenceKeys.length === 0) {
```

Comment:
Looks like we're currently double `map`ping.  I'd either check `if (references.length === 0)` or reuse `referenceKeys` online 2335.

Reply:
Ooops, the edit got lost in some branch shuffling — thanks for catching that!